plugins/inputs/opcua/opcua.go

```diff
@@ -312,7 +313,15 @@ func (o *OpcUA) validateOPCTags() error {
 
 		//search identifier type
 		switch node.tag.IdentifierType {
-		case "s", "i", "g", "b":
+		case "s":
+			if reflect.ValueOf(node.tag.Identifier).Kind() != reflect.String {
+				return fmt.Errorf("identifier type '%s' does not match the type of identifier '%s'", node.tag.IdentifierType, node.tag.Identifier)
+			}
```

Comment:
How about using a type assertion? Anyway doesn't the [definition of `Identifier`](https://github.com/influxdata/telegraf/blob/1f58747f83ab18249614610553f80409f170ef1b/plugins/inputs/opcua/opcua.go#L73) guarantee a `string`?

Reply:
I do think it has to be a string or there will be a toml error. what do you think about removing this check all together?

Reply:
I think it's safe to remove.

Reply:
done :)